libraries/ESP8266WiFiMesh/src/TcpIpMeshBackend.h

```diff
@@ -0,0 +1,283 @@
+  int32_t getConnectionAttemptTimeout();
+
+  /**
+   * Set the timeout to use for transmissions when this TcpIpMeshBackend instance acts as a station (i.e. when connected to another AP).
+   * This will affect the timeout of the attemptTransmission method once a connection to an AP has been established.
+   * The timeout is 5 000 ms by default.
+   * 
+   * @param stationModeTimeoutMs The timeout to use, in milliseconds.
+   */
+  void setStationModeTimeout(int stationModeTimeoutMs);
+  int getStationModeTimeout();
+
+  /**
+   * Set the timeout to use for transmissions when this TcpIpMeshBackend instance acts as an AP (i.e. when receiving connections from other stations).
+   * This will affect the timeout of the acceptRequests method.
+   * The timeout is 4 500 ms by default.
+   * Will also change the setting for the active AP (without an AP restart)
+   * if this TcpIpMeshBackend instance is the current AP controller.
+   *
+   * @param apModeTimeoutMs The timeout to use, in milliseconds.
+   */
+  void setAPModeTimeout(uint32_t apModeTimeoutMs);
+  uint32_t getAPModeTimeout();
+
+protected:
+
+  static std::vector<TcpIpNetworkInfo> _connectionQueue;
+  static std::vector<TransmissionOutcome> _latestTransmissionOutcomes;
+
+  /**
+   * Called just before we activate the AP.
+   * Put _server.stop() in deactivateAPHook() in case you use _server.begin() here.
+   */
+  void activateAPHook() override;
+
+  /**
+   * Called just before we deactivate the AP.
+   * Put _server.stop() here in case you use _server.begin() in activateAPHook().
+   */
+  void deactivateAPHook() override;
+
+  /** 
+   * Will be true if a transmission initiated by a public method is in progress.
+   */
+  static bool _tcpIpTransmissionMutex;
+
+  /** 
+   * Will be true when the connectionQueue should not be modified.
+   */
+  static bool _tcpIpConnectionQueueMutex;
+
+  /**
+   * Check if there is an ongoing TCP/IP transmission in the library. Used to avoid interrupting transmissions.
+   * 
+   * @return True if a transmission initiated by a public method is in progress.
+   */
+  static bool transmissionInProgress();
+
+  /**
+   * Set a message that will be sent to other nodes when calling attemptTransmission, instead of the regular getMessage(). 
+   * This message is used until clearTemporaryMessage() is called.
+   * 
+   * @param newMessage The message to send.
+   */
+  void setTemporaryMessage(const String &newMessage);
+  String getTemporaryMessage();
+  void clearTemporaryMessage();
+
+private:
+
+  uint16_t _serverPort;
+  WiFiServer _server;
+  uint8_t _maxAPStations = 4; // Only affects TCP/IP connections, not ESP-NOW connections
+  int32_t _connectionAttemptTimeoutMs = 10000;
+  int _stationModeTimeoutMs = 5000; // int is the type used in the Arduino core for this particular API, not uint32_t, which is why we use int here.
+  uint32_t _apModeTimeoutMs = 4500;
+
+  static String _temporaryMessage;
```

Comment:
Is it necessary to have private static members? it is usually better practice to have global static vars in an anonymous namespace in the .cpp (reduces clutter in the class interface and reduces export of internals).

Also, statics and non-statics should be kept apart (bool useStaticIP is sort of hidden among the statics)